zip/zipfiles.nim

```diff
@@ -159,11 +161,13 @@ iterator walkFiles*(z: var ZipArchive): string =
 
 proc extractFile*(z: var ZipArchive, srcFile: string, dest: Stream) =
   ## extracts a file from the zip archive `z` to the destination stream.
+  var buf: array[BufSize, byte]
   var strm = getStream(z, srcFile)
-  while true:
-    if not strm.atEnd:
-        dest.write(strm.readStr(1))
-    else: break
+  while not strm.atEnd:
```

Comment:
Seems dangerous. If 'atEnd' is buggy and always returns false (infinite stream?) then this turns into an endless loop. Better to use

```nim
while true:
   let bytesRead = strm.readData(addr(buf[0]), buf.len)
   if bytesRead <= 0: break
   dest.writeData(addr(buf[0]), bytesRead)
```